### DISCUSSION TITLE
WebRTC to SIP call

### DISCUSSION THREAD

**Discussion by MarkCiliaVincenti** [sipsorcery-org/sipsorcery#854] [Q&A]
Hi,

I have some code where I basically establish a WebRTC connection via websockets, send some audio, then start playing music whilst concurrently making a SIP call. Once the call is established, I'd like for it to be connected to the webRTC so someone from the web page is now talking to someone on a physical phone.

However if I `await ua.InitiateCallAsync(callDescriptor, webCall.RTCPeerConnection, timeout);` where `webCall.RTCPeerConnection` is the current RTCPeerConnection used for the WebRTC, nothing happens, the call does not go through and I get no errors or warnings.

Only if I create a new VoIPMediaSession am I able to start the call, but then I'd need to stay sending packets from the VoIPMediaSession to the RTCPeerConnection and vice versa, so I'm wondering if there's a neater way of doing it.

**Comment by MarkCiliaVincenti** [comment]
@sipsorcery any idea please?

**Comment by sipsorcery** [comment]
@MarkCiliaVincenti I may be missing something but I don't think what you're attempting is likely to work. The WebRTC RTP packets are encrypted and possibly using different codec and/or formats that the SIP user agent cannot support.

Are you using the same SDP on the WebRTC call and the SIP call? If not how are you making sure the media types match up? Are you hand crafting/hard coding the SDP to amke sure it works?

The [SIPtoWebRTCBridge](https://github.com/sipsorcery-org/sipsorcery/blob/master/examples/WebRTCExamples/SIPToWebRtcBridge/Program.cs) is the closest example to what you're attempting and it does forward RTP packets between the two sessions rather than trying to create a single session between the SIP and WebRTC clients.

All that's not to say a SIP User Agent can't directly establish a media session with a WebRTC peer, providing you sort the signalling out, but it would be tricky. sip.js, see [example](https://github.com/sipsorcery-org/sipsorcery/blob/master/examples/WebRTCExamples/sipjs/sipphone.html), attempts to do something along those lines but it requires SIP over Web Sockets and I think you'd end up running into the same problem you mention above for most remote SIP UA's.
